### ISSUE TITLE
Cannot set custom Tile properties with GameObjectCreator#tilemap

### ISSUE THREAD

**Issue by iamchristopher** [photonstorm/phaser#3267]
* A bug in the API (always say which version you're using!)
v3.1.1

I'm working on a project that takes individual rooms created in Tiled and randomly stitches them together to create a larger map for the player to explore. My current implementation goes something like this:

```js
const tilemap = this.make.tilemap({
    data: [], // Array containing tile indices
    tileWidth: 32,
    tileHeight: 32
});
const tileset = tilemap.addTilesetImage('tiles');
const layer = tilemap.createDynamicLayer(0, tileset, 0, 0);

const collisionMap = []; // Collision generated from map tiles
this.impact.world.setCollisionMap(collisionMap, 32);
```

This all works great. However, now I'd like to get into some more advanced collision detection after reading about [setting collision by property](http://labs.phaser.io/view.html?src=src%5Cgame%20objects%5Ctilemap%5Cstatic%5Cset%20colliding%20by%20property.js&v=114). After digging through the source code I've come to the conclusion that this functionality only works when reading from cache data and not when passing in custom data.

Is there a workaround for this or is it currently a limitation with `GameObjectCreator#tilemap`?

**Comment by mikewesthad** [comment]
Hey @iamchristopher - you can manually set properties on your Tile objects by manipulating `Tile#properties`. Then you can use `setCollisionByProperty(...)` however you want.

When creating a tilemap from an array of indices, there are no tile properties in the data to speak of. It's just indices. The array format is intentionally super simple. If you want something more complicated, you could create a blank tilemap layer and then manipulate the tiles directly by looping over them and setting indices/properties/colliding however you want.

**Comment by iamchristopher** [comment]
Thanks for the response, @mikewesthad! I had a sneaking suspicion that would be the solution. Thanks for confirming :)

**Comment by iamchristopher** [comment]
@mikewesthad Before I move to implement your recommended solution, I just wanted to clarify.. You said this format is intentionally simple--does that mean you wouldn't be open to a PR adding that functionality?

**Comment by mikewesthad** [comment]
Would you be able to explain a bit more about the use case you envision? I'm having trouble picturing why the array format api needs to be more complicated vs creating a blank map and using the Tile api. 

On a second read of your situation, since you've got this modular Tiled setup, I imagine you could automate things nicely. You could loop over all the tiles in your map and use `Tileset#getTileProperties` (if all the rooms share a tileset) to copy over the props you set up in Tiled.

**Comment by iamchristopher** [comment]
Sure, I'll try to explain..

1. I've created a handful of room segments (8x10) in Tiled. Segments are typed by direction (left, right, and down)
2. Given a room dimension (ie, 6x6 segments) I generate a room layout by creating a 2D array containing indices representing segment types. This guarantees a path across the room while creating alternate paths that are dead ends
3. I map over this layout and insert segment (layer) data from Tiled, giving me a 2D array containing room segments
4. Next I need to merge all of the rooms together to create a single massive map that can be used as a Tilemap
5. Finally I can create a Collision Map using the merged data

From a visual standpoint this all works wonderfully (props to all the work that's been done with Tilemaps et al), but now that I'm trying to use this generated Tilemap like a regular one exported from Tiled by accessing properties I've hit a wall.

I tried using `Tileset#getTileProperties` as you suggested and came up empty as the Tileset doesn't seem to have any properties. My assumption is that because the Tileset is created using the generated Tilemap (`addTilesetImage`) it isn't aware of any tile properties.

Your idea of simply creating a blank map sounds interesting. Is there an example of this technique available? I've found [this one](http://labs.phaser.io/edit.html?src=src\game%20objects\tilemap\dynamic\creating%20blank%20layers.js) but it randomizes the layout instead of having a fixed one. I'll dig around in the meantime and see if I can figure something out. This way seems like less of a hack 😁
